### PR TITLE
tests: some integration test UX enhancements

### PR DESCRIPTION
**Pull request by orndorffgrant**
no-jira no-lp no-gh
Some ideas for improvements I had during verification testing.
(well there are lots more ideas too, but these were small enough I went ahead and did them).

## Test Steps

Try out user data for configuring a test run. e.g. `tox -e behave-lxd-16.04 -- -D install_from=archive`
Cause a test failure and then check out the contents of `./artifacts/latest/logs`

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by renanrodrigo — APPROVED** [review]
hello hello
`black` says goodbye I say hello

File: features/environment.py [hunk 1]
```diff
@@ -496,31 +470,19 @@ def after_step(context, step):
...
-                with open(artifact_file, "w") as stream:
-                    stream.write(content)
-            for artifact_file, cmd in FAILURE_CMDS.items():
-                result = context.machines[SUT].instance.execute(
-                    cmd, use_sudo=True
-                )
-                artifact_file = os.path.join(artifacts_dir, artifact_file)
-                with open(artifact_file, "w") as stream:
-                    stream.write(result.stdout)
+            context.machines[SUT].instance.execute(
```
Comment (lucasmoura) [inline features/environment.py:473]: Should we have a `try..except` block for the execute commands ? I am worried if we fail here somehow we might not cleanup the machines after (even though we have the cleanup scripts)

Comment (orndorffgrant) [reply features/environment.py:473]: yep good point I'll add that

**Review by lucasmoura — APPROVED** [review]

**Comment by orndorffgrant** [comment]
CI fails are unrelated, merging

**Pull request merged**